internal/services/compute/migration/snapshot_v0_to_v1.go

```diff
@@ -0,0 +1,85 @@
+package migration
+
+import (
+	"context"
+
+	"github.com/hashicorp/go-azure-sdk/resource-manager/compute/2022-03-02/snapshots"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/pluginsdk"
+)
+
+var _ pluginsdk.StateUpgrade = SnapshotV0ToV1{}
+
+type SnapshotV0ToV1 struct{}
+
+func (SnapshotV0ToV1) UpgradeFunc() pluginsdk.StateUpgraderFunc {
+	return func(ctx context.Context, rawState map[string]interface{}, meta interface{}) (map[string]interface{}, error) {
+		oldId, err := snapshots.ParseSnapshotIDInsensitively(rawState["id"].(string))
+		if err != nil {
+			return rawState, err
+		}
+
+		rawState["id"] = oldId.ID()
+		return rawState, nil
+	}
+}
```

Comment:
can we also remove the old Resource ID Type from `resourceids.go` and the generated ID Parsers/Validators?